rules/xcconfig.doc.bzl

```diff
@@ -0,0 +1,10 @@
+load(
+    "//rules/library:xcconfig.bzl",
```

Comment:
Should this live under `rules/library/xcconfig.doc.bzl` to be next to its associated rule? I presume we don't want all of `//rules/library:xcconfig.bzl` to be considered public API, so we're cherry-picking a subset of it here? Should we mark the rest as internal symbols with an underscore prefix or something, if that prevents it from being documented?

Same feedback for `vfs_overlay`.

Reply:
The use of `.doc.bzl` matches what `rules_apple` does, it's exporting the public api of `xcconfig.bzl` in the `//rules` package specifically. This makes it so we can create the `stardoc` target for these in one list vs. conditionalizing the path to the `bzl_library`

See this in `docs/BUILD.bazel`:

```
stardoc(
   ...
   input = "//rules:%s.bzl" % file,
)
```